core/src/main/java/com/airbnb/aerosolve/core/transforms/NgramTransform.java

```diff
@@ -0,0 +1,118 @@
+package com.airbnb.aerosolve.core.transforms;
+
+import com.airbnb.aerosolve.core.FeatureVector;
+import com.airbnb.aerosolve.core.util.Util;
+
+import java.util.ArrayList;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import com.typesafe.config.Config;
+
+/**
+ * Tokenizes strings using a regex and generates ngrams from the tokens
+ * "field1" specifies the key of the feature
+ * "regex" specifies the regex used to tokenize
+ * "n" specifies the size of the ngrams
+ * "min_n" optional parameter, if specified ngrams from min_n (inclusive) to n (inclusive) will
+ *  be generated and placed in the output
+ */
+public class NgramTransform implements Transform {
```

Comment:
since this is new, we may move it under pkg string, and use it like string.ngram  
this categorizes the Transform from now on.